pip/_vendor/requests/packages/urllib3/packages/ssl_match_hostname/_implementation.py

```diff
@@ -108,14 +108,14 @@ def match_hostname(cert, hostname):
     try:
         # Divergence from upstream: ipaddress can't handle byte str
         host_ip = ipaddress.ip_address(_to_unicode(hostname))
-    except ValueError:
-        # Not an IP address (common case)
-        host_ip = None
     except UnicodeError:
         # Divergence from upstream: Have to deal with ipaddress not taking
         # byte strings.  addresses should be all ascii, so we consider it not
         # an ipaddress in this case
         host_ip = None
+    except ValueError:
```

Comment:
unfortunately this is a change in a vendored package, please submit it upstream, as otherwise with the next update of requests pip would loose this enhancement